code/iaas/logic/src/main/resources/META-INF/cattle/core-process/spring-core-processes-context.xml

```diff
@@ -144,4 +144,25 @@
     <!-- Cluster host map -->
     <process:process name="clusterhostmap.create" resourceType="clusterHostMap" start="requested" transitioning="creating" done="created" />
     <process:process name="clusterhostmap.remove" resourceType="clusterHostMap" start="created,creating" transitioning="removing" done="removed" />
+
+    <!-- Service Discovery Environment -->
+    <process:process name="environment.create" resourceType="environment" start="requested" transitioning="activating" done="active" />
+    <process:process name="environment.activateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
+    <process:process name="environment.deactivateservices" resourceType="environment" start="active" transitioning="updating-active" done="active" />
```

Comment:
@sonchang The reason why I've put the state, is to prevent activate/deactivate services to be called in parallel. Keeping the state should be harmless I guess.